File: src/main/java/de/neo/eventsystem/EventSystemTest.java
package de.neo.eventsystem;

public class EventSystemTest {

    public static void main(String[] args) throws InterruptedException {
        EventSystem eSys = new EventSystem();
        eSys.registerListener(new TestListener());
        eSys.registerListener(new LowTestListener());
        eSys.executeEvent(new TestEvent("Test 1"));
        Thread.sleep(5000);
        eSys.executeEvent(new TestEvent("Test 2"));
    }
}

class TestEvent extends Event {

    private String msg;

    public TestEvent(String msg) {
        this.msg = msg;
    }

    public String getMessage() {
        return this.msg;
    }
}

class TestListener {

    @Listener
    public void onTest(TestEvent event) {
        System.out.println("Neue Nachricht: " + event.getMessage());
    }

}

class LowTestListener {

    @Listener
    public void onTest(TestEvent event) {
        System.out.println("Neue Nachricht (low): " + event.getMessage());
    }

}

File: src/main/kotlin/de/neo/eventsystem/kt/EventSystemTest.kt
package de.neo.eventsystem.kt

fun main() {
    val eSys = EventSystem()
    eSys.registerListener(TestListener())
    eSys.registerListener(LowTestListener())
    eSys.executeEvent(TestEvent("Test 1"))
    Thread.sleep(5000)
    eSys.executeEvent(TestEvent("Test 2"))
}

class TestEvent(val msg: String) : Event() {
}

class TestListener {

    @Listener
    fun onTest(event: TestEvent) {
        println("Neue Nachricht: ${event.msg}")
    }

}

class LowTestListener {

    @Listener
    fun onTest(event: TestEvent) {
        println("Neue Nachricht (low): ${event.msg}")
    }

}